### DISCUSSION TITLE
Popover full width for mobile hamburger menu

### DISCUSSION THREAD

**Discussion by isshou-tora** [adobe/react-spectrum#6648] [Q&A]
Hi folks, I am trying to build mobile menu by using the sample provided in this page [https://react-spectrum.adobe.com/react-aria/Menu.html](url). 
When I set the popover width to 100%, it exceeds the viewport width:
![image](https://github.com/adobe/react-spectrum/assets/131500425/c4b3b2a7-10f7-4758-a5da-c5d13ba00e8a)
I saw there is a "left" value in the stylesheet generated by the popover:
![image](https://github.com/adobe/react-spectrum/assets/131500425/63be596b-7e07-4b9e-8e7e-263d2d54f8e5)
Is there a way to set/override it to 0px?
Thank you in advance

**Comment by NickWoodward** [comment]
Yeah, I found that style really annoying too. It doesn't seem possible to override it with `!important` either, presumably because they've already marked their inline style that way.

Do you want the menu to fill the entire width? Or do you want it to not overflow?

- If you don't want it to overflow, remove the `width: 100%` and add `margin-right: 12px` to even it out.
- If you want it to take up the whole width, luckily they haven't used the translate property, so you can: add `transform: translate(-12px, 0px)`.

Hope that helps, and hope someone comes in with a better way to deal with that property, because it (maybe naively) seems kind of silly.

Nick

**Reply by little-chili** [reply]
Thanks Nick, I want it to take the whole width.
I am just wondering whether the left will always be 12px. Anyway, I will try your suggestion.
Thanks a lot

**Comment by little-chili** [comment]
Using containerPadding={0} can remove the left value
